wasm/jit/jit_amd64_test.go

```diff
@@ -2948,6 +2948,780 @@ func TestAmd64Compiler_compilPopcnt(t *testing.T) {
+							{x1Value: 2, x2Value: 1},
+							{x1Value: 1, x2Value: 2},
+							{x1Value: 0, x2Value: 1},
+							{x1Value: 1, x2Value: 0},
+							{x1Value: 0, x2Value: 0},
+							// Following cases produce different resulting bit patterns for signed and unsigned.
+							{x1Value: 0xffffffffffffffff /* -1 in signed 64bit */, x2Value: 1},
+							{x1Value: 0xffffffffffffffff /* -1 in signed 64bit */, x2Value: 0xfffffffffffffffe /* -2 in signed 64bit */},
+						} {
+							vs := vs
+							t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+
+								eng := newEngine()
+								compiler := requireNewCompiler(t)
+								compiler.initializeReservedRegisters()
+
+								// Pretend there was an existing value on the DX register. We expect compileDivForInts to save this to the stack.
+								// Here, we put it just before two operands as ["any value used by DX", x1, x2]
+								// but in reality, it can exist in any position of stack.
+								compiler.movIntConstToRegister(int64(dxValue), x86.REG_DX)
+								prevOnDX := compiler.locationStack.pushValueOnRegister(x86.REG_DX)
+
+								// Setup values.
+								if tc.x1Reg != -1 {
+									compiler.movIntConstToRegister(int64(vs.x1Value), tc.x1Reg)
+									compiler.locationStack.pushValueOnRegister(tc.x1Reg)
+								} else {
+									loc := compiler.locationStack.pushValueOnStack()
+									eng.stack[loc.stackPointer] = uint64(vs.x1Value)
+								}
+								if tc.x2Reg != -1 {
+									compiler.movIntConstToRegister(int64(vs.x2Value), tc.x2Reg)
+									compiler.locationStack.pushValueOnRegister(tc.x2Reg)
+								} else {
+									loc := compiler.locationStack.pushValueOnStack()
+									eng.stack[loc.stackPointer] = uint64(vs.x2Value)
+								}
+
+								var err error
+								if signed.signed {
+									err = compiler.compileDiv(&wazeroir.OperationDiv{Type: wazeroir.SignedTypeInt64})
+								} else {
+									err = compiler.compileDiv(&wazeroir.OperationDiv{Type: wazeroir.SignedTypeUint64})
+								}
+								require.NoError(t, err)
+
+								require.Equal(t, int16(x86.REG_AX), compiler.locationStack.peek().register)
+								require.Equal(t, generalPurposeRegisterTypeInt, compiler.locationStack.peek().regType)
+								require.Equal(t, uint64(2), compiler.locationStack.sp)
+								require.Len(t, compiler.locationStack.usedRegisters, 1)
+								// At this point, the previous value on the DX register is saved to the stack.
+								require.True(t, prevOnDX.onStack())
+
+								// We add the value previously on the DX with the quotiont of the division result
+								// in order to ensure that not saving existing DX value would cause
+								// the failure in a subsequent instruction.
+								err = compiler.compileAdd(&wazeroir.OperationAdd{Type: wazeroir.UnsignedTypeI64})
+								require.NoError(t, err)
+
+								// The division by zero error must be caught by Go's runtime via x86's exception caught by kernel.
+								defer func() {
+									if e := recover(); e != nil {
```

Comment:
extract this func so you don't need to copy/pasta and also you can re-use the comment as godoc?